test/Host/TestUtils/RepoDirectoriesProvider.cs

```diff
@@ -31,7 +31,7 @@ public class RepoDirectoriesProvider
             string corehostPackages = null,
             string corehostDummyPackages = null)
         {
-            _repoRoot = repoRoot ?? Path.Combine(Directory.GetCurrentDirectory(), "..", "..");
+            _repoRoot = repoRoot ?? Path.Combine(Directory.GetCurrentDirectory(), "..", "..", "..");
```

Comment:
I don't know how to get rid of CWD, but to get rid of the `..`s this could be changed to look at parent directories until it hits one that "looks" like the repo root due to some file being present. (Similar to typical `dir.props` or `dir.targets` directory traversal.) `branchinfo.txt` seems like a decent target.